test/sample-test/run_resnet_cmle_test.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import json
+import tarfile
+from datetime import datetime
+import utils
+from kfp import Client
+
+###### Input/Output Instruction ######
+# input: yaml
+# output: local file path
+
+
+# Parsing the input arguments
+def parse_arguments():
+  """Parse command line arguments."""
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--input',
+                      type=str,
+                      required=True,
+                      help='The path of a pipeline package that will be submitted.')
+  parser.add_argument('--result',
+                      type=str,
+                      required=True,
+                      help='The path of the test result that will be exported.')
+  parser.add_argument('--output',
+                      type=str,
+                      required=True,
+                      help='The path of the test output')
+  parser.add_argument('--namespace',
+                      type=str,
+                      default='kubeflow',
+                      help="namespace of the deployed pipeline system. Default: kubeflow")
+  args = parser.parse_args()
+  return args
+
+def main():
+  args = parse_arguments()
+  test_cases = []
+  test_name = 'Resnet CMLE Test'
+
+  ###### Initialization ######
+  host = 'ml-pipeline.%s.svc.cluster.local:8888' % args.namespace
+  client = Client(host=host)
+
+  ###### Check Input File ######
+  utils.add_junit_test(test_cases, 'input generated yaml file', os.path.exists(args.input), 'yaml file is not generated')
+  if not os.path.exists(args.input):
+    utils.write_junit_xml(test_name, args.result, test_cases)
+    print('Error: job not found.')
+    exit(1)
+
+  ###### Create Experiment ######
+  experiment_name = 'resnet cmle sample experiment'
+  response = client.create_experiment(experiment_name)
+  experiment_id = response.id
+  utils.add_junit_test(test_cases, 'create experiment', True)
+
+  ###### Create Job ######
```

Comment:
Is it possible to generalize the test script and only pass the `job_name` and `params`?

Reply:
I'll leave it to @gaoning777 :).

Reply:
potentially it could. The reason of having multiple scripts was different checking logic for each sample. Since the sample tests generally checks whether the run succeeds(except the kubeflow_test checks the confusion matrix), we could refactor the codes.